drafts/25-xxx-specifications.txt

```diff
@@ -413,22 +413,141 @@ in28. Unlike INCLUDE lines (see the section on "INCLUDE lines"),
       the #include directive is not prohibited from including the
       same source file at a deeper level of nesting.
 
+
 3.5 The '#if', '#ifdef', '#ifndef', '#elif', '#elifdef',
-      '#elifndef', '#else', '#endif' directives
+      '#elifndef', '#else', '#endif' conditional directives
 ---------------------------------------------------------
 
-Example syntax:
+Example syntax (extra spacing for illustration purposes only):
+    General form:
+        <if-head>  <lines>                        #endif EOL
+        <if-head>  <lines>  <if-elif>             #endif EOL
+        <if-head>  <lines>  <if-elif>  <if-else>  #endif EOL
+
+    Where:
+        <if-head> is one of:
+            #if token-list EOL  <lines>
+            #ifdef ID EOL       <lines>
+            #ifndef ID EOL      <lines>
+
+        <if-elif> is zero or more of:
+            #elif     token-list EOL  <lines>
+            #elifdef  ID EOL          <lines>
+            #elifndef ID EOL          <lines>
 
+        <if-else> is zero or one of:
+            #else EOL  <lines>
+
+        <lines> are zero or more of:
+            directive lines
+            Fortran comment lines
+            Fortran source lines
 
 
 3.5.1 Static semantics specifications
---------------------------------------
+-------------------------------------
+
+if05. A #if, ##ifdef, or #ifndef directive signals the start of a list
+      of conditional directives.
+
+if10: Whitespace must immediately follow the directives tokens
+      'ifdef', 'ifndef', 'elifdef', and 'elifndef' to separate them
+      from the ID tokens.
+
+if15. The list of #elif, #elifdef, and #elifndef directives, and the
+      #else directive (if present) are part of the same list of
+      directives introduced by the #if directive.
+
+if20. The chain of conditional directives ends with the #endif
+      directive.
+
+if25. Within a conditional directive list, a #if, #ifdef, or #endif
+      directive introduces a new list of conditional directives, at a new
+      nesting level, within the enclosing conditional directive list.
+
+if30. The conditional directive lists must properly nest.
+
+if35. As shown in the illustrative syntax, there should be no
+      unmatched #endif directives.
```

Comment:
This property can be directly derived from the pseudo-grammar above, so repeating it here is redundant. Worse, we only state one such property but omit others, for example: 

1. there should be no `<if-head>` without a matching `#endif`, 
2. there should be no `<if-elif>` without a matching `<if-head>` and `#endif`, 
3. there should be no `#else` without a matching `<if-head>` and `#endif`, 

which makes this one seem like an outlier. We should either list all such analogous grammatical properties or none.

Reply:
I'm OK with letting the grammar speak for itself here, as long as we believe J3 as a whole won't feel like it's under-specified.

Reply:
Remove if35.

Reply:
Made changes in today's update. Will request further reviews.